### DISCUSSION TITLE
In the `with-docker-multi-env` example why are the development and staging sample `.env` files both copied into `env.production` rather than referencing the development or staging environment name?

### DISCUSSION THREAD

**Discussion by philwolstenholme** [vercel/next.js#45738] [Help]
Hi,

I'm wondering why the [`with-docker-multi-env`](https://github.com/vercel/next.js/tree/canary/examples/with-docker-multi-env) example copies the sample `.env` files into `.env.production` in the `staging` and `development` Dockerfiles rather than `.env.staging` and `.env.development`?

According to https://nextjs.org/docs/basic-features/environment-variables#environment-variable-load-order these files could be called `.env.$(NODE_ENV).local` or `.env.$(NODE_ENV)`.

Would a `.env.production` file be loaded on the `staging` environment, for example?

I can't work out if the example Dockerfiles need to be edited, or if I am missing something? 

---

[Staging](https://github.com/vercel/next.js/blob/canary/examples/with-docker-multi-env/docker/staging/Dockerfile#L23-L24):

```dockerfile
# This will do the trick, use the corresponding env file for each environment.
COPY .env.staging.sample .env.production
```

[Development](https://github.com/vercel/next.js/blob/canary/examples/with-docker-multi-env/docker/development/Dockerfile#L22-L23):

```dockerfile
# This will do the trick, use the corresponding env file for each environment.
COPY .env.development.sample .env.production
```

**Comment by deadcoder0904** [comment]
same question.

i thought this was answered.

@philwolstenholme did you find out the answer?

**Reply by deadcoder0904** [reply]
@samcx can you explain this since you wrote the code for this?

**Reply by samcx** [reply]
@deadcoder0904 I have not written the code for this.

Feel free to push up a PR to fix this!
